libbeat/_meta/config.reference.yml

```diff
@@ -245,6 +245,12 @@ output.elasticsearch:
   # IPv6 addresses should always be defined as: https://[2001:db8::1]:9200
   hosts: ["localhost:9200"]
 
+  # Enabled ilm to use index lifecycle management instead daily indices.
+  #ilm.enabled: false
+  #ilm.rollover_alias: "beat-index-prefix"
+  # This default pattern is {now/d}-000001 escaped as it contains date math
+  #ilm.pattern: "%7Bnow%2Fd%7D-000001"
```

Comment:
WDYT about moving the `ilm` config options to the `setup.template` section? The options are stored within the template so it makes sense to move them over. 
With https://github.com/elastic/beats/pull/9247 we will be able to have multiple templates per beat, which allows to set different policies per template. This also requires a `ilm.rollover_alias` per template. Having this in mind, moving the `ilm` config to `settings.template` would make a lot of sense.

Reply:
I can see the argument that it belongs into `setup` as most steps for ILM configuration are part of the setup. But the template modification is only one part. One thing we could do:

```
output.elasticsearch.ilm.enabled: true
setup.ilm.rollover_alias: "foo"
setup.ilm.pattern: "bar"
```

I think the ilm enabled makes sense in the output as it overwrites the existing index settings.

Reply:
Thinking about having multiple templates, how would you connect a `rollover_alias` to a template? If we could do something like this for now: 
```
setup.template:
  name: "metricbeat"
  pattern: "metricbeat*"
  ilm.enabled: true
  ilm.rollover_alias: "metricbeat"
  ilm.pattern: "000001"
```
and once the multiple templates per beat is merged change it to this, it would be relatively straight forward. The thing I haven't thought through is whether or not `ilm.enabled` should be set per template or not. 
```
setup.template.templates:
  - name: "metricbeat-foo"
    pattern: "*"
    modules: "aerospike,apache,system"
    overwrite: true
    ilm.enabled: true
    ilm.rollover_alias: "metricbeat"
    ilm.pattern: "000001"
  - name: "metricbeat-docker"
    pattern: "docker*"
    modules: "docker"
    enabled: true 
    overwrite: true
    ilm.enabled: true
    ilm.rollover_alias: "metricbeat"
    ilm.pattern: "000001"
```

Reply:
The reason I'm focusing on ILM instead of the template is that from my perspective the template is more a side effect of ILM and what the user wants to do is configure ILM and not necessarily the template. I would rather nest optional template under ILM but I would actually hope that the template part can directly be deducted from the ILM config.

Let's assume we have something like this:
```
setup.ilm:
  # This is the same as the index the beat needs to write to
  rollover_alias: "apm-%{[beat.version]}-%{processor.event}"
  ilm.pattern: "000001"
  vars:
    processor.event:
    - error
    - span
    - transaction
```

This would create 3 templates and would do the magic of pattern and name for the user. But it does not offer the flexibility you tried above with having 3 modules to one pattern.

Having the above, the user could actually set the index to `output.elasticsearch.index: "apm-%{[beat.version]}-%{processor.event}"` and things should just work. Should we set this automatically when ilm is enabled and require the user not to set an index, meaning we don't start if one is set so we can set it and not only overwrite it?

Let's keep iterating on this one I think we haven't found that optimal solution yet.

To keep the PR moving, would you be ok to move forward with the current options but do follow up PR's on it?

Reply:
I am fine with that, as long as we do not release a version with the current config options, as we would need to support them then and be backwards compatible if we want to change them.

Reply:
I think we will find a solution to the problem before the release. But even if it would be release, we are still in beta stage and could still change the config options if needed.